fastjet/internal/delaunay/point.go

```diff
@@ -50,13 +51,21 @@ func (p *Point) inTriangle(t *Triangle) (inside, edge bool) {
 	barcen2 := ((t.C.Y-t.A.Y)*(p.X-t.C.X) + (t.A.X-t.C.X)*(p.Y-t.C.Y)) / det(t)
 	barcen3 := 1 - barcen1 - barcen2
 
-	// inside triangle
-	if barcen1 > 0 && barcen1 < 1 && barcen2 > 0 && barcen2 < 1 && barcen3 > 0 && barcen3 < 1 {
+	// use the math/big package to handle the geometric predicates
```

Comment:
My original idea was that there could be several implementations of the predicates with various levels of robustness vs speed tradeoff. It could be configurable and the user could even provide its own external implementation. This is of course not necessary and you can just implement these as two unexported functions.